app/portainer/home/EnvironmentList/KubeconfigPrompt.tsx

```diff
@@ -0,0 +1,151 @@
+import { useState } from 'react';
+import { useQuery } from 'react-query';
+
+import * as kcService from '@/kubernetes/services/kubeconfig.service';
+import * as notifications from '@/portainer/services/notifications';
+import { Button } from '@/portainer/components/Button';
+import { Checkbox } from '@/portainer/components/form-components/Checkbox';
+import { EnvironmentId } from '@/portainer/environments/types';
+import { EnvironmentsQueryParams } from '@/portainer/environments/environment.service/index';
+import { isKubernetesEnvironment } from '@/portainer/environments/utils';
+import { PaginationControls } from '@/portainer/components/pagination-controls';
+import { usePaginationLimitState } from '@/portainer/hooks/usePaginationLimitState';
+import { useEnvironmentList } from '@/portainer/environments/queries/useEnvironmentList';
+
+import styles from './KubeconfigPrompt.module.css';
+import '@reach/dialog/styles.css';
+
+function useSelection() {
+  const [selection, setSelection] = useState<Record<EnvironmentId, boolean>>(
+    {}
+  );
+
+  const selectionSize = Object.keys(selection).length;
+
+  return { selection, toggle, selectionSize };
+
+  function toggle(id: EnvironmentId, selected: boolean) {
+    setSelection((prevSelection) => {
+      const newSelection = { ...prevSelection };
+
+      if (!selected) {
+        delete newSelection[id];
+      } else {
+        newSelection[id] = true;
+      }
+
+      return newSelection;
+    });
+  }
+}
+
+export interface KubeconfigPromptProps {
+  envQueryParams: EnvironmentsQueryParams;
+  onToggleClose: () => void;
+}
+const storageKey = 'home_endpoints';
+
+export function KubeconfigPrompt({
+  envQueryParams,
+  onToggleClose,
+}: KubeconfigPromptProps) {
+  const [page, setPage] = useState(1);
+  const [pageLimit, setPageLimit] = usePaginationLimitState(storageKey);
+  const kubeServiceExpiryQuery = useQuery(['kubeServiceExpiry'], async () => {
+    const expiryMessage = await kcService.expiryMessage();
+    return expiryMessage;
+  });
+  const { selection, toggle: toggleSelection, selectionSize } = useSelection();
+  const { environments, totalCount } = useEnvironmentList({
+    page,
+    pageLimit,
+    ...envQueryParams,
+  });
+  const kubeEnvs = environments.filter((env) =>
+    isKubernetesEnvironment(env.Type)
+  );
```

Comment:
same comment as before, 
```suggestion
  const { environments, totalCount } = useEnvironmentList({
      ...envQueryParams, // moved to top so other params will replace those in the queryParams
    page,
    pageLimit,
    types: [...kube env types]
  });
  const kubeEnvs = environments.filter((env) =>
    isKubernetesEnvironment(env.Type)
  );
```

Reply:
Done